File: Elixir/mastermind/lib/mastermind.ex
defmodule Mastermind do
  @moduledoc """
  Functions for playing the game Mastermind
  """

  def generate_code() do
    [:blue, :blue, :yellow, :red]
  end

  def guess([]), do: :error

  def guess(colours) do
    [:black, :black, :black, :black]
  end
end

File: Ruby/src/mastermind.rb
class Mastermind

  COLOURS = [:red, :blue, :green, :orange, :purple, :yellow]

  def initialize
    @secret = []
  end

  def secret
    @secret = COLOURS.sample(4)
  end

  def secret=(secret)
    @secret = secret
  end

  def guess(guess)
    result = []
    guess.each_with_index do |colour, index|
      if @secret[index] == colour
        result << :black
      elsif @secret.include?(colour)
        result << :white
      end
    end

    result
  end
end
